Changes: The tab title to Menu from Site Menu

AjeshRPai opened the pull request:
Fixes #16305 

This PR changes the title of the tab from `Site menu` to `Menu` as per [this reference](pbArwn-4jn). 

![Screenshot_20220412_192436](https://user-images.githubusercontent.com/17463767/162980405-53cfd562-bbe4-4538-a0de-154078bf957f.png)


To test:

Test 1 
- Go to My Site 
- Notice that the title of the tabs are `Home` and `Menu` 

Test 2
- Go to My Site -> App Settings -> Initial Screen 
- Notice the selections are `Home` and `Menu`

Test 3 
- Go to My Site screen with a Site having Quick Start in Progress 
- Set Initial screen as Home 
- Start any quick start(Visit Site, Explore Plans or Post Sharing) which has focus point in `Menu` tab
- Notice that the snackbar prompt says to click at the `Menu` tab

## Regression Notes
1. Potential unintended areas of impact
Tab title is not shown properly in 
- Initial screen settings 
- Quick start snack bar prompt 


2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing 

3. What automated tests I added (or what prevented me from doing so)
none 

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

AjeshRPai committed:
Changes: The tab title to Menu from Site Menu

zwarm reviewed (changes requested):
👋  @AjeshRPai - The tab names look great, as does the App Settings -> Initial Screen selection view screen; however there seems to be a mismatch between "site menu" and "menu" in the label. When "menu" is the default, it is not showing properly as the label in the App Settings view. This was the same issue I encountered because "dashboard" and "home" do not match.

Pre-req: Have to be in a situation where the "home" tab is assigned by the experiment.
- Fresh install
- Login
- Select site
- Navigate to Me -> App Settings
- Note that the Initial Screen setting does not have a value. 
- <img width="250" height="500" alt="Alt desc" src="https://user-images.githubusercontent.com/506707/162989215-80c775e6-e451-4a8f-9f73-15518e49e98d.png">

AjeshRPai committed:
Updates: Mapping of variant tracking

AjeshRPai commented:
@zwarm Great catch, Thanks for pointing it out and letting me know how to reproduce it. I believe I have made the necessary changes in the commit [60a71ef](https://github.com/wordpress-mobile/WordPress-Android/pull/16307/commits/60a71efeb01917c8faa42113812e563d9eefc28b). I have tested the scenario where the `Menu` tab is assigned by the experiment. 

![Screenshot_20220413_131338](https://user-images.githubusercontent.com/17463767/163128554-b12c57b9-a4c5-4c28-be78-e93211f2a323.png)

zwarm reviewed (changes requested):
Hi @AjeshRPai . The issue with App Settings look good, but now there is an issue selecting the "menu" tab. 

Steps to reproduce
- Navigate to Me -> App Settings
- Set initial screen to "menu"
- Close the app (swipe it away)
- Open the app
- "Menu" should be the default tab, but "home" is selected instead

Take a look in MySiteViewModel ... The initial screen value = "menu" which doesn't == "site_menu".  
```
private val defaultABExperimentTab: MySiteTabType
        get() = if (isMySiteTabsEnabled) {
           if (appPrefsWrapper.getMySiteInitialScreen() == MySiteTabType.SITE_MENU.label) {
                MySiteTabType.SITE_MENU
            } else {
                MySiteTabType.DASHBOARD
            }
        } else {
            MySiteTabType.ALL
        }
```

AjeshRPai committed:
Updates: Logic of the preference key being stored and default value

AjeshRPai commented:
@zwarm Yup, Right. How did I miss that? 😓 . I have done the changes again in the commit [2757860](https://github.com/wordpress-mobile/WordPress-Android/pull/16307/commits/27578600929ce84deede4b5e21a8ed8152c9a0f7) and tested the following scenario. It works as expected. 

- Navigate to Me -> App Settings
- Set initial screen to "menu"
- Close the app (swipe it away)
- Open the app
- 🟢 "Menu" should be the default tab

I hope that I haven't missed any other scenarios now. 🤞🏼 . Please have a look at the changes.

ashiagr committed:
Fix failing tests

ashiagr committed:
Fix redundant qualifier warning

ashiagr committed:
Pass correct value to setInitialScreenFromMySiteDefaultTabExperimentVariant

1. It seems we were comparing initial screen label with variant tracking label on preference change. Since site menu label is changed to menu, this commit adjusts comparison of changed preference value with the correct value.

2. APP_SETTINGS_INITIAL_SCREEN_CHANGED -> "selected" value tracks variant tracking labels (dashboard, site_menu). It matches corresponding tracking in iOS.

3. Passing variant tracking label to setInitialScreenFromMySiteDefaultTabExperimentVariant via AppSettingsFragment.onPreferenceChange and initial screen value from MySiteDefaultTabExperiment.checkAndSetVariantIfNeeded looked incorrect. This commit passes variant tracking label from both places to setInitialScreenFromMySiteDefaultTabExperimentVariant() where it gets converted to the initial screen label (reflecting function's true purpose).

ashiagr committed:
Rename my site tab label to tracking label

This is done as current my site tab labels match the variant tracking labels: dashboard, site_menu
Note that tracking label will be replaced with the actual label when creating new tab fragment instance in the next commit.

ashiagr committed:
Add my site tab label

My site tab labels now match the actual labels: home, menu
Intermediary tracking label used when creating a new tab fragment can now be replaced with this label.

ashiagr committed:
Replace VARIANT_DASHBOARD, VARIANT_DASHBOARD with my site tab tracking labels

VARIANT_DASHBOARD, VARIANT_DASHBOARD were used to track experiment variants. They now match my site tab tracking labels and can be replaced with them.

ashiagr committed:
Replace VARIANT_HOME, VARIANT_MENU with my site tab labels

VARIANT_HOME, VARIANT_MENU were used to save initial screen value in app prefs. They now match my site tab labels and can be replaced with them.

ashiagr committed:
Replace home, menu in tests with my site tab labels

ashiagr commented:
Hey @zwarm! 

As @AjeshRPai is on leave till the end of this week, I took over this PR to fix failing tests.

I tested it t to see if everything was working when I noticed an issue with changing the initial screen preference: the initial screen preference value was compared with the site menu variant tracking label which are different values now (80c187a5ad51b11913aab22cf3ff3adf56aec963). In order to fix it, I refactored `AppPrefs.setInitialScreenFromMySiteDefaultTabExperimentVariant()` slightly with details in the commit message. I hope I understood the function's purpose correctly 🤞 else let's sync to get this right. 

I also eliminated a few constants (`VARIANT_DASHBOARD`, `VARIANT_DASHBOARD`, `VARIANT_HOME`, `VARIANT_MENU`, `home`, `menu`) at different places in commits ([95b1dfd](https://github.com/wordpress-mobile/WordPress-Android/pull/16307/commits/95b1dfdc923b5d7588c727302a21779d4e1d68a3) ... [47dca0a](https://github.com/wordpress-mobile/WordPress-Android/pull/16307/commits/47dca0a303502572538ecf64349f64c67bb8a739)). If you feel these are too many changes, I can revert them. 😅 

Thanks so much for your time on this. 🙏

ashiagr committed:
Replace new pref value with track value

zwarm reviewed (approved):
@ashiagr - All works as expected with the changes you made. The correct "tracking label" is used for tracking. The initial screen value is updated properly and the app setting is correct. And the tabs names look great!!
Thanks for getting this one over the finish line. 
🎉
